server/tests/db/test_connection_pool.py

```
@@ -0,0 +1,261 @@
+    cursor.__enter__ = MagicMock(return_value=cursor)
+    cursor.__exit__ = MagicMock(return_value=False)
+    connection = MagicMock(name="connection")
+    connection.cursor.return_value = cursor
+    return connection, cursor
+
+
+def _executed_sql(cursor):
+    return [c.args[0] for c in cursor.execute.call_args_list if c.args]
+
+
+@pytest.fixture()
```

Ruff (0.15.12)

[warning] Use `@pytest.fixture` over `@pytest.fixture()`

Remove parentheses

(PT001)